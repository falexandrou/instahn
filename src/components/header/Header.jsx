import React from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

/**
 * Our Header Navigational bar
 */
class Header extends React.Component {
  static propTypes = {
    isMenuOpen: PropTypes.bool.isRequired,
    onMenuToggled: PropTypes.func.isRequired,
  };

  static defaultProps = {
    isMenuOpen: false,
  };

  render() {
    const { isMenuOpen, onMenuToggled } = this.props;

    return <nav className="navbar is-primary is-transparent is-flex" role="navigation" aria-label="main navigation">
      <div className="navbar-brand">
        <a className="navbar-item hackernews-logo" href="/">
          <img src="/assets/hackernews.svg" alt="Hackernews" />
        </a>
      </div>

      <button onClick={ onMenuToggled } className={`button navbar-burger is-block-desktop is-primary is-inverted is-outlined ${isMenuOpen ? 'is-active' : '' }`}>
        <span />
        <span />
        <span />
      </button>
    </nav>;
  }
}

export default Header;