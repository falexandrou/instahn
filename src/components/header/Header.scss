.hackernews-logo {
  color: #fff;
}

.navbar {
  margin-bottom: 2em;

  .navbar-burger.is-primary.is-inverted.is-outlined {
    border: none;

    &:hover, &:focus, &:active {
      border: none;
      background: transparent;
      color: #fff;
      box-shadow: none;
    }
  }
}

