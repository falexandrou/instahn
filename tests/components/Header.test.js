import 'raf/polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from 'components/header/Header';

describe('Header Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('should match snapshot', () => {
    const output = renderer.create(<Header />).toJSON();
    expect(output).toMatchSnapshot();
  });
});