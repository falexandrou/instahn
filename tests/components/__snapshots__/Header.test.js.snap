// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-primary is-transparent is-flex"
  role="navigation"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item hackernews-logo"
      href="/"
    >
      <img
        alt="Hackernews"
        src="/assets/hackernews.svg"
      />
    </a>
  </div>
  <button
    className="button navbar-burger is-block-desktop is-primary is-inverted is-outlined "
    onClick={undefined}
  >
    <span />
    <span />
    <span />
  </button>
</nav>
`;
